### DISCUSSION TITLE
Linuxbrew dependency bug in brew pr-upload for taps

### DISCUSSION THREAD

**Discussion by jeroen** [Homebrew/discussions#4340] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
(CI job)
```

### Output of `brew doctor`

```shell
(CI job)
```

### Description of issue

I have been maintaining a tap for a few years, with CI based on examples from [dawidd6/homebrew-tap](https://github.com/dawidd6/homebrew-tap).

Recently the CI job that downloads and commits the bottles seems to have broken. It looks like the problem is that the audit looks for dependencies __only__ within the tap, but does not find dependencies that are from homebrew-core.

See log file here: https://github.com/autobrew/homebrew-cran/actions/runs/4460250374/jobs/7833263066?pr=63

```
Bundle complete! 35 Gemfile dependencies, 78 gems now installed.
Bundled gems are installed into `../../home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/bundle`
==> autobrew/cran/libgit2-static
  bottle do
    root_url "https://github.com/autobrew/homebrew-cran/releases/download/libgit2-static-1.4.5"
    sha256 cellar: :any,                 monterey:     "98a0f32fe775489262735e5b7019cb33fba79da13cc0ef99f0ac534d2a11cb71"
    sha256 cellar: :any,                 big_sur:      "7d1670547fd462be1f7582c2c05831a2c4441[180](https://github.com/autobrew/homebrew-cran/actions/runs/4460250374/jobs/7833263066?pr=63#step:4:181)a767161cf928325869504f80"
    sha256 cellar: :any_skip_relocation, x86_64_linux: "b3da955985fb2ed5626718fc836fe71cf572c2f1dddbeddca1a6c4ec9143dc3d"
  end
[master 45c32b7] libgit2-static: update 1.4.5 bottle.
 1 file changed, 4 insertions(+), 5 deletions(-)
Error: 4 problems in 1 formula detected
autobrew/cran/libgit2-static:
  * Can't find dependency '"cmake"'.
  * Can't find dependency '"pkg-config"'.
Error: Can't find dependency '"cmake"'.
Error: Can't find dependency '"pkg-config"'.
Error: Can't find dependency '"cmake"'.
Error: Can't find dependency '"pkg-config"'.
Error: Failure while executing; `/home/linuxbrew/.linuxbrew/bin/brew audit --skip-style autobrew/cran/libgit2-static` exited with 1.
Error: Failure while executing; `/home/linuxbrew/.linuxbrew/bin/brew pr-upload` exited with 1.
Error: Process completed with exit code 1.
```

**Comment by dawidd6** [comment]
You can try adding the following lines below `runs-on` in you workflow:
```yaml
container:
  image: ghcr.io/homebrew/ubuntu22.04:master
```

**Reply by jeroen** [reply]
I [tried that](https://github.com/autobrew/homebrew-cran/commit/30c3de49b87b5bd32060558769d266c09506401d) but it did not make any difference. I also tried replacing all workflows the vanilla workflows from `brew tap-new` but still no success.

**Comment by carlocab** [comment, marked as answer]
Alternatively, set
```
HOMEBREW_NO_INSTALL_FROM_API
```
in your workflow.

**Reply by jeroen** [reply]
Thanks, it looks like this fixes the issue. Perhaps this variable should be included in the default templates from `brew tap new` (or better, `pr-upload` should not assume API).
